src/components/DonationCardModal/DonationCardModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import styles from './DonationCardModal.module.css';
+
+interface DonationCardModalProps {
+   title: string;
+   target: number;
+   amount: number;
+   onUpdateDonationAmount: (amount: number) => void;
+   onCloseModal: () => void;
+}
+
+function DonationCardModal({
+   title,
+   target,
+   amount,
+   onUpdateDonationAmount,
+   onCloseModal,
+}: DonationCardModalProps) {
+   const [donationAmount, setDonationAmount] = useState(0);
+
```

Comment:
General thinking:
Is this going to be the only Modal in the whole project?
If not, there could be 2 components:
- 1 that has the layout (title, xbutton, footer)
- second that have the "content" that can send to the generic Modal as children?
If you want we can talk more about it as Im not sure Im explaining it well 😅

Reply:
’ll admit, I like this idea and agree with the approach. However, I believe it will require a significant amount of refactoring, so I’m not entirely sure what to do next.

Reply:
leave it to a "future" refactor...